rustls/src/server/hs.rs

```diff
@@ -362,6 +362,19 @@ impl State for ExpectClientHello {
             ProtocolVersion::TLSv1_2
         };
 
+        if version == ProtocolVersion::TLSv1_3
+            && client_hello.client_version != ProtocolVersion::TLSv1_2
+        {
+            // RFC 8446 - 4.1.2
+            // In TLS 1.3, the client indicates its version preferences in the
+            // "supported_versions" extension (Section 4.2.1) and the
+            // legacy_version field MUST be set to 0x0303, which is the version
+            // number for TLS 1.2.
```

Comment:
This is referencing/quoting the part of the spec that requires the client to set the value to 0x0303. Instead, we should be referencing/quoting the part of the spec that requires us to fail with an alert if the client put the wrong value in. I couldn't find that part of the spec. Is this check actually required by the spec? If not, I'm inclined to not do it.